tests/manage/pv_services/test_resource_deletion_during_pvc_pod_creation_and_io.py

```diff
@@ -0,0 +1,286 @@
+            namespace=self.namespace
+        )
+        log.info("Successfully created new pods using all PVCs.")
+
+        # Updating self.pod_objs for the purpose of teardown
+        self.pod_objs.extend(pod_objs_re)
+
+        # Run IO on each of the newly created pods
+        for pod_obj in pod_objs_re:
+            pod_obj.run_io(
+                storage_type='fs', size='100M', runtime=10,
+                fio_filename='fio-file-retest'
+            )
+
+        log.info("Fetching IO results from newly created pods")
+        for pod_obj in pod_objs_re:
+            get_fio_rw_iops(pod_obj)
+        log.info("Verified IO result on newly created pods.")
+
+
+@tier4
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_ceph_block_pool.__name__,
+    create_rbd_storageclass.__name__,
+    create_pvcs.__name__,
+    test_fixture.__name__
+)
+class TestForRBD(OperationsBase):
+    """
+    Test class for RBD
+    """
+    interface = constants.CEPHBLOCKPOOL
+
+    @pytest.mark.parametrize(
+        argnames="resource_to_delete",
+        argvalues=[
+            pytest.param(
+                *['mgr'],
+                marks=pytest.mark.polarion_id("OCS-735")
+            ),
+            pytest.param(
+                *['mon'],
+                marks=pytest.mark.polarion_id("OCS-736")
+            ),
+            pytest.param(
+                *['osd'],
+                marks=pytest.mark.polarion_id("OCS-737")
+            )
+
+        ]
+    )
+    def test_resource_deletion_during_pvc_pod_creation_and_io_block(
+        self, resource_to_delete
+    ):
+        """
+        Delete resource while PVC creation, Pods creation, IO are progressing.
+        RBD PVC
+        """
+        self.operations_base(resource_to_delete)
+
+
+@tier4
+@pytest.mark.usefixtures(
+    create_cephfs_secret.__name__,
+    create_cephfs_storageclass.__name__,
+    create_pvcs.__name__,
+    test_fixture.__name__
+)
+class TestForCephFS(OperationsBase):
+    """
+    Test class for CephFS
+    """
+    interface = constants.CEPHFILESYSTEM
+
+    @pytest.mark.parametrize(
+        argnames="resource_to_delete",
+        argvalues=[
+            pytest.param(
+                *['mgr'],
+                marks=pytest.mark.polarion_id("OCS-738")
+            ),
+            pytest.param(
+                *['mon'],
+                marks=pytest.mark.polarion_id("OCS-739")
+            ),
+            pytest.param(
+                *['osd'],
+                marks=pytest.mark.polarion_id("OCS-740")
+            ),
+            pytest.param(
+                *['mds'],
+                marks=pytest.mark.polarion_id("OCS-741")
+            )
+        ]
+    )
+    def test_resource_deletion_during_pvc_pod_creation_and_io_file(
+        self, resource_to_delete
+    ):
+        """
+        Delete resource while PVC creation, Pods creation, IO are progressing.
+        CephFS PVC
+        """
+        self.operations_base(resource_to_delete)
```

Comment:
Jilju and I are meeting on Monday, 5th Aug to do a live review of this PR. I request you not to merge until then.

Reply:
Here are the comments from our meeting:
1) The script should be creating the RWX and RWO based PVCs as expected by test cases. 
-- Shreekara's PR will help here.
2) The FIO needs to verify the data integrity of the data it wrote. The PR #412 has specific yaml and small code changes to do so. 
3) Pod restarts count need to be checked before and after to make sure there is no unexpected pod restart happening.
   -- PR #412 is having this code. Harish will create a separate PR for verifying the restarts count so it can be used across all T4 tests.

Reply:
These are added to the the disruptive cases [tracker](https://github.com/red-hat-storage/ocs-ci/issues/382#issuecomment-518155983).
These steps will be added in test case after the mentioned PRs get merged.

Reply:
ok. resolving this comment